src/lib/Name.hs

```diff
@@ -3326,6 +3332,78 @@ toListNameMap :: NameMap c a n -> [(Name c n, a)]
 toListNameMap (UnsafeNameMap raw) = R.toList raw <&> \(r, x) -> (UnsafeMakeName r, x)
 {-# INLINE toListNameMap #-}
 
+unionWithNameMap :: (a -> a -> a) -> NameMap c a n -> NameMap c a n -> NameMap c a n
+unionWithNameMap f (UnsafeNameMap raw1) (UnsafeNameMap raw2) =
+  UnsafeNameMap $ R.unionWith f raw1 raw2
+{-# INLINE unionWithNameMap #-}
+
+traverseNameMap :: (Applicative f) => (a -> f b)
+                 -> NameMap c a n -> f (NameMap c b n)
+traverseNameMap f (UnsafeNameMap raw) = UnsafeNameMap <$> traverse f raw
+{-# INLINE traverseNameMap #-}
+
+mapNameMap :: (a -> b) -> NameMap c a n -> (NameMap c b n)
+mapNameMap f (UnsafeNameMap raw) = UnsafeNameMap $ fmap f raw
+{-# INLINE mapNameMap #-}
+
+newtype NameMapE (c::C) (e:: E) (n::S) = NameMapE (NameMap c (e n) n)
+  deriving (Eq, Semigroup, Monoid)
+
+hoistFilterNameMapE :: (BindsNames b, HoistableE e, ShowE e)
+                    => b n l -> NameMapE c e l -> HoistExcept (NameMapE c e n)
+hoistFilterNameMapE b (NameMapE nmap) =
```

Comment:
Well the point of `hoistFilter` was that it never fails, but rather filters out the names that fail to hoist. So please either do that, or just call it `host` without the "filter" part

Reply:
Done.